ThrownExceptionFinder acts as a caretaker for memento S, which processes TryStatement parameter of tryStatement, BlockScope parameter of scope Thrown Exceptions, which accepts ReferenceBinding parameter of binding Exception, which ends MessageSend parameter of messageSend, BlockScope parameter of scope Visit, which ends AllocationExpression parameter of allocationExpression, BlockScope parameter of scope Visit, which ends ThrowStatement parameter of throwStatement, BlockScope parameter of scope Visit, which ends MethodBinding parameter of methodBinding Visit Method Invocation, which gets Already Caught Exceptions, which gets Thrown Uncaught Exceptions, which gets Discouraged Exceptions, which visits TypeDeclaration parameter of typeDeclaration, CompilationUnitScope parameter of scope, which visits TypeDeclaration parameter of memberTypeDeclaration, ClassScope parameter of scope, which visits TypeDeclaration parameter of localTypeDeclaration, BlockScope parameter of scope, which visits TypeDeclaration parameter of typeDeclaration Type, which visits TryStatement parameter of tryStatement, BlockScope parameter of scope, which removes TryStatement parameter of tryStatement, boolean parameter of recordUncheckedCaughtExceptions Caught Exceptions, which removes ReferenceBinding parameter of caughtException Caught Exception.
